build/yamls/flow-visibility/base/clickhouse.yml

```diff
@@ -36,7 +36,100 @@ spec:
               volumeMounts:
                 - name: clickhouse-configmap-volume
                   mountPath: /docker-entrypoint-initdb.d
+                - name: clickhouse-storage-volume
+                  mountPath: /var/lib/clickhouse
           volumes:
             - name: clickhouse-configmap-volume
               configMap:
                 name: $(CLICKHOUSE_CONFIG_MAP_NAME)
+            - name: clickhouse-storage-volume
+              emptyDir:
+                medium: Memory
+                sizeLimit: 8Gi
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app: flow-visibility
+  name: clickhouse-monitor
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  labels:
+    app: flow-visibility
+  name: clickhouse-monitor-role
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+    verbs:
+      - list
+  - apiGroups:
+      - ""
+    resources:
+      - pods/log
+    verbs:
+      - get
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  labels:
+    app: flow-visibility
+  name: clickhouse-monitor-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: clickhouse-monitor-role
+subjects:
+  - kind: ServiceAccount
+    name: clickhouse-monitor
+---
+apiVersion: batch/v1
+kind: CronJob
```

Comment:
why does this run as `CronJob` which I think will schedule a new Pod each time, instead of as an additional container in the Clickhouse Pod?

Reply:
As `Cronjob` is scheduled by Kubernetes, we think in terms of lifecycle, it provides better management. 
We also consider to deploy the Clickhouse cluster in future which may have multiple Clickhouse pods, but we only need one monitor for the whole Clickhouse cluster.

Reply:
I see the consideration of the overhead. We do have some discussions in the previous team meeting about the plan to use linux crontab in a continuously running pod. 

We think the Cronjob might be more reliable. For example, when the continuously running pod goes wrong, the monitor won't work any more. But with Cronjob each pod runs separately thus the failure of one pod does not affects the others. This might be more important when the cluster is large. I'm still no sure whether we should reconsider the crontab design.

Reply:
Having some more discussions with @salv-orlando offline, I understand the reliability and saving of overhead with the single continuously running pod monitor solution. I plan to switch the current Cronjob monitor to a continuously running pod in a follow up PR.

Reply:
Agreed. The current implementation assumes something like a cronjob is running, so it's better to keep the using it. We can iterate over it in the next release (with the caveat that we will need to find a way to terminate the cronjob or document it must be terminated)